packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -904,7 +904,11 @@ class _CupertinoModalPopupRoute<T> extends PopupRoute<T> {
 ///
 /// The `useRootNavigator` argument is used to determine whether to push the
 /// popup to the [Navigator] furthest from or nearest to the given `context`. It
-/// is `false` by default.
+/// is `true` by default.
+///
+/// The `useBarrierColor` argument is used to degermine whether to use barrier
```

Comment:
Thanks for your contribution. This argument as-is is a bit single purposed. It's better to just let the user specify their own barrierColor (which could be null if that's needed). See the `showDialog` static function's arguments.

Reply:
@xster Sure, I will fix it. 

But is it convenient to use `CupertinoDynamicColor.resolve(_kModalBarrierColor, context)` as a default value for this parameter. I mean with `null` value as acceptable parameter value we can't fall back to `CupertinoDynamicColor.resolve(_kModalBarrierColor, context)`, as it must be the default, back-ward compatible behavior. What do you think?

Reply:
@xster Pushed the next iteration of the idea. Personally I am not quite sure about it but can't invent anything better.